src/Abs/Download/PartialFileDownloader.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\FileStorage\Abs\Download;
+
+use Keboola\FileStorage\FileNotFoundException;
+use Keboola\FileStorage\Compression\CompressionDetectorHelper;
+use Keboola\FileStorage\Path\RelativePathInterface;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+use MicrosoftAzure\Storage\Blob\Models\GetBlobOptions;
+use MicrosoftAzure\Storage\Common\Exceptions\ServiceException;
+use MicrosoftAzure\Storage\Common\Models\Range;
+use Symfony\Component\Process\Process;
+
+final class PartialFileDownloader
+{
+    private const ERR_CODE_INVALID_PAGE_RANGE = 416;
+    private const ERR_CODE_BLOB_NOT_FOUND = 404;
+    private const BYTES_RANGE_START = 0;
+    public const BYTES_RANGE_END = 655360;
+
+    /** @var BlobRestProxy */
+    private $blobClient;
+
+    public function __construct(BlobRestProxy $blobClient)
+    {
+        $this->blobClient = $blobClient;
+    }
+
+    public function downloadBytes(RelativePathInterface $path, int $bytes = self::BYTES_RANGE_END): string
+    {
+        $tmpFilePath = null;
+        $options = new GetBlobOptions();
+
+        $options->setRange(new Range(self::BYTES_RANGE_START, $bytes));
+
+        try {
+            $result = $this->blobClient->getBlob(
+                $path->getRoot(),
+                $path->getPathnameWithoutRoot(),
+                $options
+            );
+        } catch (ServiceException $e) {
+            if ($e->getCode() === self::ERR_CODE_BLOB_NOT_FOUND) {
+                throw new FileNotFoundException($path, $e);
+            }
+            if ($e->getCode() === self::ERR_CODE_INVALID_PAGE_RANGE) {
+                // file is empty
+                return '';
+            }
+
+            throw $e;
+        }
+
+        if (CompressionDetectorHelper::isGzipped($path->getPathnameWithoutRoot())) {
+            /** @var string $tmpFilePath */
+            $tmpFilePath = tempnam(sys_get_temp_dir(), 'PartialFileDownloader');
+            file_put_contents($tmpFilePath, $result->getContentStream());
+            $process = Process::fromShellCommandline(sprintf('zcat %s', $tmpFilePath));
+            $process->mustRun();
+            $body = $process->getOutput();
+            unlink($tmpFilePath);
```

Comment:
Na tohle chybí test. A zkusil jsem si to nasimulovat bokem a přijde mi že `zcat` neumí pracovat s těma partial souborama?

Reply:
test doplním, ale taky je to copy paste z https://github.com/keboola/connection/blob/master/Package/FileStorage/src/FileHeaderDownloader/AbsFileHeaderDownloader.php

Reply:
Udělal sem update, fakt to failne když je ten gzip větší než range, takže sem to změnil. V connection je to tímpádem taky blbě :)